packages/shared/src/pipelines.ts

```diff
@@ -0,0 +1,95 @@
+export interface Operation<T, D = unknown> {
+  invoke(data: D): T;
+}
+
+export class Pipeline<T, D> implements Operation<T[]> {
```

Comment:
If we're going to introduce these new functional primitives, can we make sure they're clearly documented?

Reply:
If we are going to introduce them i will add JSDoc with examples for each primitive :)